rotkehlchen/exchanges/coinbase.py

```diff
@@ -278,68 +222,72 @@ def validate_api_key(self) -> tuple[bool, str]:
-            if result is None:
-                return False, msg
-
-            # and now try to get all sells of an account to see if that's possible
-            method = f'accounts/{account_ids[0]}/sells'
+            method = f'accounts/{account_info[0][0]}/transactions'
             result, msg = self._validate_single_api_key_action(method)
             if result is None:
                 return False, msg
 
             # and now try to get all deposits of an account to see if that's possible
-            method = f'accounts/{account_ids[0]}/deposits'
+            method = f'accounts/{account_info[0][0]}/deposits'
             result, msg = self._validate_single_api_key_action(method)
             if result is None:
                 return False, msg
 
             # and now try to get all withdrawals of an account to see if that's possible
-            method = f'accounts/{account_ids[0]}/withdrawals'
+            method = f'accounts/{account_info[0][0]}/withdrawals'
             result, msg = self._validate_single_api_key_action(method)
             if result is None:
                 return False, msg
 
         return True, ''
 
-    def _get_account_ids(self, accounts: list[dict[str, Any]]) -> list[str]:
-        """Gets the account ids out of the accounts response"""
-        account_ids = []
+    def _get_active_account_info(self, accounts: list[dict[str, Any]]) -> list[tuple[str, Timestamp]]:  # noqa: E501
+        """Gets the account ids and last_update timestamp out of the accounts response
+
+        Only returns the active ones. We assume that if created_at and updated_at are
+        the same then the account is not active. This is important as the API requires
+        iteration over all accounts to get history and there is A LOT of them.
+        """
+        account_info = []
         for account_data in accounts:
             if 'id' not in account_data:
-                self.msg_aggregator.add_error(
+                log.error(
                     'Found coinbase account entry without an id key. Skipping it. ',
                 )
                 continue
 
             if not isinstance(account_data['id'], str):
-                self.msg_aggregator.add_error(
+                log.error(
                     f'Found coinbase account entry with a non string id: '
                     f'{account_data["id"]}. Skipping it. ',
                 )
                 continue
 
-            account_ids.append(account_data['id'])
+            updated_at = account_data.get('updated_at')
+            try:
+                timestamp = deserialize_timestamp_from_date(updated_at, 'iso8601', 'coinbase')
+            except DeserializationError:
+                log.error(f'Skipping coinbase account {account_data} due to inability to deserialize timestamp')  # noqa: E501
+                continue
+
+            log.debug(f'Found coinbase account: {account_data}')
+
+            if account_data.get('created_at') == updated_at:
```

Comment:
This `if` saves **a lot** of time. Can't find the assumption documented in their API docs, but their API docs are terribly outdated and don't even have half the transaction types I used in coinbase and I am not a heavy user at all.

```suggestion
            if account_data.get('created_at') == updated_at:
```